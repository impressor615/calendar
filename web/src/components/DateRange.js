import React from 'react';
import PropTypes from 'prop-types';
import DatePicker from 'react-datepicker';

const DateRange = ({ startDate, endDate, onChange }) => (
  <div className="date-range">
    <DatePicker
      showTimeSelect
      selected={startDate}
      onChange={onChange}
      timeIntervals={60}
      timeCaption="time"
      dateFormat="YYYY-M-D h:mm A"
    />
    <div>~</div>
    <div>{ endDate.format('YYYY-M-D h:mm A') }</div>
  </div>
);

DateRange.propTypes = {
  startDate: PropTypes.object.isRequired,
  endDate: PropTypes.object.isRequired,
  onChange: PropTypes.func.isRequired,
}

export default DateRange;
