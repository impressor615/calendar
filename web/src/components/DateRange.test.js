import React from 'react';
import moment from 'moment';
import DatePicker from 'react-datepicker';

import DateRange from 'components/DateRange';


describe('<DateRange />', () => {
  const onChange = jest.fn()
  const startDate = moment('2018-10-13');
  const endDate = moment('2018-10-13').add(1, 'hours');
  const props = {
    onChange,
    startDate,
    endDate,
  };

  it('should be rendered properly', () => {
    const wrapper = shallow(<DateRange {...props} />);
    expect(wrapper.find('.date-range')).toHaveLength(1);
    expect(wrapper.find(DatePicker)).toHaveLength(1);
  });

  it('should be matched with the snapshot', () => {
    const wrapper = shallow(<DateRange {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});


