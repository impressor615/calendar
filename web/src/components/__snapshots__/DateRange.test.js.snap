// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DateRange /> should be matched with the snapshot 1`] = `
<div
  className="date-range"
>
  <DatePicker
    allowSameDay={false}
    dateFormat="YYYY-M-D h:mm A"
    dateFormatCalendar="MMMM YYYY"
    disabled={false}
    disabledKeyboardNavigation={false}
    dropdownMode="scroll"
    monthsShown={1}
    nextMonthButtonLabel="Next month"
    onBlur={[Function]}
    onChange={[MockFunction]}
    onClickOutside={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMonthChange={[Function]}
    onSelect={[Function]}
    onYearChange={[Function]}
    preventOpenOnFocus={false}
    previousMonthButtonLabel="Previous Month"
    readOnly={false}
    selected={"2018-10-12T15:00:00.000Z"}
    shouldCloseOnSelect={true}
    showTimeSelect={true}
    timeCaption="time"
    timeIntervals={60}
    withPortal={false}
  />
  <div>
    ~
  </div>
  <div>
    2018-10-13 1:00 AM
  </div>
</div>
`;
